### PR TITLE
utils: Misc fixes

### PR DESCRIPTION
**Pull request by LeSpocky**
Minor fixes which accumulated when hacking on different things. Patches 1 and 2 are self contained. Patch 4 depends on patch 3, those two could be squashed if desired.

### CODE REVIEW

**Comment by LeSpocky** [comment]
The last change which just affected documentation let the code coverage test fail. I have no idea why. 🤷🏼‍♂️

**Comment by ejoerns** [comment]
Hm, not sure how it concludes that total coverage in main.c decreased which is not touched at all. But this is not a blocker for merging, anyway.

File: include/utils.h [hunk 1]
```diff
@@ -134,6 +134,8 @@ G_GNUC_WARN_UNUSED_RESULT;
 
 /**
  * Get string argument from key and remove key from key_file.
+ *
+ * @return A newly allocated string or NULL.
```
Comment (ejoerns) [inline include/utils.h:138]: Maybe even more clearer with:

``` suggestion
 * @return A newly allocated string or NULL on error.
```

In a follow up we should then also add documentation for the parameters, but I would not see that as the blocker for this initial enhancement ;)

**Review by ejoerns — APPROVED** [review]
:+1:

**Review by jluebbe — APPROVED** [review]

**Pull request merged**